src/core/utilities/span.h

```diff
@@ -15,17 +15,44 @@ struct Span
 {
     using value_type = T;
 
-    T& operator[](SIZE i) const { return ptr[i]; }
+    auto& operator[](SIZE i) { return ptr[i]; }
+    auto& operator[](SIZE i) const { return ptr[i]; }
     T const* const& data() const { return ptr; }
     T const* const& begin() const { return ptr; }
     T* end() const { return ptr + s; }
     SIZE const& size() const { return s; }
 
-
     T const* ptr = nullptr;
     SIZE s       = 0;
 };
 
+
+template<typename T, typename SIZE = size_t>
+class VectorSpan : public StackVar<std::vector<T>>, public core::Span<T, SIZE>
+{
+    using Super0 = StackVar<std::vector<T>>;
```

Comment:
not fan of super0/1 ...
why not Span and Vector or something?

Reply:
sure